src/components/map/Map.tsx

```diff
@@ -157,131 +144,264 @@ export const Map = ({
+            setIsInitialFit(false); // Verhindert zukünftige automatische Anpassungen
+          }
+        }
+      }
+    }
+  }, [locations, clusterDistance, zoomLevel, isInitialFit, clusteringEnabled]);
+
+  useEffect(() => {
+    if (updatedRegion && cameraRef.current) {
+      cameraRef.current.setCamera({
+        centerCoordinate: [updatedRegion.longitude, updatedRegion.latitude],
+        zoomLevel: 14,
+        animationDuration: 1000
+      });
+    }
+  }, [updatedRegion]);
+
+  const handleMapPress = (event) => {
+    if (onMapPress) {
+      const { geometry } = event;
+      const nativeEvent = {
+        coordinate: {
+          latitude: geometry.coordinates[1],
+          longitude: geometry.coordinates[0]
+        }
+      };
+      onMapPress({ nativeEvent });
+    }
+  };
+
+  const handleClusterPress = useCallback(
+    (cluster) => {
+      if (isAnimating) return; // Skip if animation is already in progress
+
+      setIsAnimating(true);
+      const [longitude, latitude] = cluster.geometry.coordinates;
+      const zoom = Math.min(zoomLevel + 2, 20); // Increase the zoom level by 2 or set to max zoom level (20)
+      if (cameraRef.current) {
+        cameraRef.current.setCamera({
+          centerCoordinate: [longitude, latitude],
+          zoomLevel: zoom,
+          animationDuration: 1000
+        });
+      }
+
+      // Wait for the animation to finish
+      setTimeout(() => {
+        setIsAnimating(false);
+      }, 1000);
+    },
+    [isAnimating, zoomLevel]
+  );
+
   return (
     <View style={[styles.container, style]}>
-      <MapView
-        clusterColor={colors.primary}
-        clusterFontFamily="regular"
-        clusteringEnabled={clusteringEnabled}
-        radius={clusterDistance}
-        renderCluster={renderCluster}
-        initialRegion={initialRegion}
-        region={updatedRegion}
-        mapType={device.platform === 'android' ? MAP_TYPES.NONE : MAP_TYPES.STANDARD}
-        onPress={onMapPress}
+      <LoadingSpinnerMap loading={isLoading} />
+      <MapLibreGL.MapView
+        style={mapStyle || styles.map}
+        styleJSON="https://tileserver-gl.smart-village.app/styles/osm-liberty/style.json"
+        onPress={handleMapPress}
         ref={refForMapView}
-        rotateEnabled={false}
-        showsBuildings={false}
-        showsPointsOfInterest={false}
-        showsUserLocation={showsUserLocation}
-        showsMyLocationButton={false}
-        toolbarEnabled={false}
-        style={[stylesForMap().map, mapStyle]}
-        userLocationPriority="balanced"
-        minZoom={minZoom}
-        mapPadding={{
-          top: 0,
-          right: 0,
-          bottom: -normalize(30),
-          left: 0
-        }}
-        legalLabelInsets={{
-          top: 0,
-          right: 0,
-          bottom: normalize(10),
-          left: 0
+        onRegionDidChange={(region) => {
+          const newZoomLevel = Math.round(region.properties.zoomLevel);
+          if (!isNaN(newZoomLevel)) {
```

Comment:
this checks if `newZoomLevel` is a number? this reads very hard "is not not a number"...
i suggest checking the type of the variable on "number" like explained at https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/isNaN

> [Number.isNaN()](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Number/isNaN) is a more reliable way to test whether a value is the number value NaN or not. Alternatively, the expression x !== x can be used, and neither of the solutions is subject to the false positives that make the global isNaN() unreliable. To test if a value is a number, use [typeof x === "number"](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/typeof).

Reply:
thanks, it will be changed in the upcoming commit to: 

`if (typeof newZoomLevel === 'number' && !isNaN(newZoomLevel)) {...`

Reply:
both? the hard to understand double negation is still present for what reason?

Reply:
should it just be: `if (!isNaN(newZoomLevel)) {...` ?

Reply:
no, this was you first approach which caused the discussion, right?!

Reply:
instead of adding `typeof newZoomLevel === 'number'` you should have exchanged it

Reply:
exchanged for what? I don't get what you want me to do rn.

Reply:
exchange the difficult !NaN... with `typeof newZoomLevel === 'number'`... `if (typeof newZoomLevel === 'number')...`